gpiozero/pins/mock.py

```diff
@@ -413,3 +385,34 @@ def tx_word(self, value, bits_per_word=None):
             bits = reversed(bits)
         self.tx_buf.extend(bits)
 
+
+class MockFactory(LocalPiFactory):
+    def __init__(self, revision='a21041', pin_class=MockPin):
+        super(MockFactory, self).__init__()
+        self._revision = revision
+        self.pin_class = pin_class
+
+    def _get_address(self):
+        return ('mock',)
+
+    def _get_revision(self):
+        return self._revision
+
+    def reset(self):
```

Comment:
Must admit I hadn't considered running the mock pin tests on a real Pi (I figured running the real pin tests should be sufficient, assuming we can achieve decent coverage on them). But it's an interesting idea. I think for now I'm going to err on the side of not adding too much - if we move reset up the hierarchy later it won't be an API break as it'll just mean adding a method to the other factories. I'd like to get the "real" pin tests cleaned up a bit before we tackle adding more stuff to them (and there's some mock tests that I doubt we'll ever support on real pins; e.g. testing pins where frequency is unsupported).